src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -82,28 +92,29 @@ static ClassDesc of(String name) {
     }
 
     /**
-     * Returns a {@linkplain ClassDesc} for a class or interface type,
-     * given the name of the class or interface in internal form,
-     * such as {@code "java/lang/String"}.
-     *
-     * @apiNote
-     * To create a descriptor for an array type, either use {@link #ofDescriptor(String)}
-     * or {@link #arrayType()}; to create a descriptor for a primitive type, use
-     * {@link #ofDescriptor(String)} or use the predefined constants in
-     * {@link ConstantDescs}.
+     * {@return a {@linkplain ClassDesc} from a string representation
+     * in a {@link ##constant-class-info CONSTANT_Class_info} structure}
      *
-     * @param name the fully qualified class name, in internal (slash-separated) form
-     * @return a {@linkplain ClassDesc} describing the desired class
+     * @param name the class name, compliant with requirements of the
+     * {@code CONSTANT_Class_info} structure
      * @throws NullPointerException if the argument is {@code null}
      * @throws IllegalArgumentException if the name string is not in the
      * correct format
-     * @jvms 4.2.1 Binary Class and Interface Names
      * @see ClassDesc#of(String)
      * @see ClassDesc#ofDescriptor(String)
+     * @see ClassDesc#internalName()
+     * @see <a href="#constant-class-info">The {@code CONSTANT_Class_info} Structure</a>
      * @since 20
```

Comment:
This should have `@revised`, as array descriptors are not allowed as input to this method in **JDK 20**.
```suggestion
     * @since 20
     * @revised 21
```

Reply:
There's no guideline on using revised. Also, I don't think we will declare it revised if it starts accepting Valhalla Q-types.